File: 1375. Bulb Switcher III/Solution_TLE.py
OFF = 0
ON = 1
BLUE = 2

class Solution:
    def numTimesAllBlue(self, light: List[int]) -> int:
        vStates = [0] * len(light)
        iTimes = 0
        
        for i in light:
            vStates[i-1] = ON
            # print(vStates)
            
            iOnMax = -1
            for i,v in enumerate(vStates):
                if v > 0:
                    iOnMax = i
                else:
                    break
            # print(iOnMax)
            
            for i in range(0, iOnMax + 1):
                vStates[i] = BLUE
            
            if 2 in vStates and 1 not in vStates:
                iTimes += 1
            
        return iTimes
            

File: 1375. Bulb Switcher III/Solution_TLE.cpp
#define OFF 0
#define ON 1
#define BLUE 2

class Solution {
public:
    bool onlyBlue(vector<int>& light)
    {
        for (auto &v : light)
        {
            if (v == 1)
                return false;
        }
        return true;
    }
    
    
    int numTimesAllBlue(vector<int>& light) {
        vector<int> vStates(light.size(), 0);
        int iMoment = 0;
        
        for (auto &v : light)
        {
            // 1: Toggle the light
            vStates[v-1] = ON;
            
            // 2: Check if all previous bulbs are on too
            int iOnUntil = -1;
            
            for (int i=0;i<vStates.size();i++)
            {
                if (vStates[i])
                {
                    iOnUntil += 1;
                }
                else
                {
                    break;
                }
            }
            
            if (iOnUntil >= 0)
            {
                for(int i=0;i<iOnUntil+1;i++)
                {
                    vStates[i] = BLUE;
                }
            }
            if (onlyBlue(vStates))
                iMoment++;
        }
        
        return iMoment; 
        
    }
};
